frontend/public/components/pod-logs.jsx

```diff
@@ -5,59 +5,70 @@ import { Dropdown, LoadingInline, ResourceLog, ResourceName } from './utils';
 
 // Component to container dropdown or conatiner name if only one container in pod.
 const ContainerDropdown = ({currentContainer, containers, kind, onChange}) => {
-  const resourceName = (container) => <ResourceName name={container.name || <LoadingInline />} kind={kind} />;
+  const resourceName = (container) => {
+    const name = container.name || <LoadingInline/>;
+    return <ResourceName name={name} kind={kind} />;
+  };
   const dropdownItems = _.mapValues(containers, resourceName);
-  return <Dropdown className="btn-group" items={dropdownItems} title={resourceName(currentContainer)} onChange={onChange} />;
+  return <Dropdown
+    className="btn-group"
+    items={dropdownItems}
+    title={resourceName(currentContainer)}
+    onChange={onChange} />;
 };
 
 export class PodLogs extends React.Component {
   constructor(props) {
     super(props);
-    this._selectContainer = this._selectContainer.bind(this);
-
     this.state = {
       containers: [],
-      currentContainer: null,
+      currentKey: ''
     };
   }
 
   static getDerivedStateFromProps(nextProps, prevState) {
     const newState = {};
     const containers = _.get(nextProps.obj, 'status.containerStatuses', []);
-    newState.containers = _.map(containers, (container) => {
+    newState.containers = _.reduce(containers, (accumulator, {name, state}, index) => {
       return {
-        name: container.name,
-        eof: !_.isEmpty(container.state.terminated)
+        ...accumulator,
+        [name]: {
+          alive: _.has(state, 'running'),
+          name,
+          order: index
+        }
       };
-    });
+    }, {});
 
-    newState.currentContainer = prevState.currentContainer || newState.containers[0];
-    if( !_.isEqual(prevState.currentContainer, newState.currentContainer)
-        || !_.isEqual(prevState.containers, newState.containers)) {
-      return newState;
+    // Make sure current container still exists in container collection, otherwise,
+    // reset to first.
```

Comment:
It's not clear to me when this would happen. I believe the containers for a pod are immutable. Is this something we've seen?

Reply:
Yeah, you are probably right. My basis for adding this check is probably flawed. I thought that maybe there would be a scenario where a container could be added/removed to a pod, but I think you are right, that would just cause a new deployment to be triggered and a new pod to be created.